ui/src/pages/version/components/forms/components/DeploymentConfigPanel.js

```diff
@@ -53,6 +54,7 @@ export const DeploymentConfigPanel = ({
   onChange,
   errors = {},
   isEnvironmentDisabled = false,
+  setMaxAllowedReplica,
```

Comment:
can this be retrieved via environment context instead of passing as input, since they are in environment api now (if im not mistaken)

Reply:
Mmm do you mean like instead of using a setter to set the state, make all other components that are dependent on this `MaxAllowedReplica` value use the environments API to retrieve that value directly?

If so, the environments context contains all the configs for all environments and we still need to filter the correct environment that is selected in this component.